File: expandable/src/main/java/ir/drax/expandable/StateObservable.java
package ir.drax.expandable;


import java.util.Observable;

public class StateObservable extends Observable {
    private Boolean state;
    public StateObservable(Boolean state) {
        this.state=state;
    }

    public void setState(Boolean state) {
        this.state = state;
        setChanged();
        notifyObservers(state);
    }

    public Boolean getState() {
        return state;
    }
}

File: expandable/src/main/java/ir/drax/expandable/StateObservable.kt
package ir.drax.expandable

import java.util.*

class StateObservable(default:Boolean) : Observable(){
    var state:Boolean=false
        set(newState) {
            field=newState
            setChanged()
            notifyObservers(newState)
        }
    init {
        state=default
    }
}